### PR TITLE
[KEYCLOAK-19311] Distx IT tests improvements

### PR DESCRIPTION
**Pull request by andreaTP**
All of the improvements introduced for Testcontainers without Testcontainers!

### CODE REVIEW

**Review by DGuhr — DISMISSED** [review]
~~LGTM~~ gave it another shot and there's a Bug i observed when running all the tests: When doing a "Run all Tests" in intellij, the testStartDev-Test in `HelpCommandTest`always fails. That's bc. the assertHelp also gets the logmsg we're adding in `RawKeycloakDistribution.java:165` and as a result the comparison fails.  This doesn't happen when running only the `HelpCommandTest` or its `testStartDev`testmethod isolated.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,6 +5,11 @@ on: [push, pull_request]
 env:
   DEFAULT_JDK_VERSION: 11
 
+concurrency:
```
Comment (andreaTP) [inline .github/workflows/ci.yml:8]: This will stop old CIs to run even if there are new commits (and save space etc.).

cc. @stianst @abstractj

File: quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java [hunk 2]
```diff
@@ -76,17 +76,17 @@ public void afterEach(ExtensionContext context) throws Exception {
...
+            dist.stop();
+            dist.close();
         }
         super.afterAll(context);
     }
 
     private KeycloakDistribution createDistribution(DistributionTest config) {
-        KeycloakDistribution distribution = new KeycloakDistribution();
-
-        distribution.setReCreate(!ReInstall.NEVER.equals(config.reInstall()));
```
Comment (pedroigor) [inline quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java:87]: Could you please elaborate why this config is no longer set to the raw distribution?

**Comment by abstractj** [comment]
@andreaTP could you please squash the 25 commits in a single commit?

File: quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+            Path distRootPath = tmpDir.resolve("kc-tests");
+            File distFile = new File("../../../distribution/server-x-dist/target/keycloak.x-" + Version.VERSION_KEYCLOAK + ".zip");
+            if (!distFile.exists()) {
+                throw new RuntimeException("Distribution archive " + distFile.getAbsolutePath() +" doesn't exists");
+            }
+            distRootPath.toFile().mkdirs();
+            String distDirName = distFile.getName().replace("keycloak-server-x-dist", "keycloak.x");
+            Path distPath = distRootPath.resolve(distDirName.substring(0, distDirName.lastIndexOf('.')));
+
+            if (distPath.toFile().exists()) {
```
Comment (pedroigor) [inline quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:262]: We can not assume tests always want to re-create the dist. There are quite different test scenarios that should require:

* Never reinstall
* Reinstall before all
* Reinstall before each test

A good example is a test the has different steps (test methods) for building and then starting the server using the persited config.

Can we keep the way it was before?

File: quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+    private final List<String> outputStream = new ArrayList<>();
+    private final List<String> errorStream = new ArrayList<>();
+    private boolean manualStop;
+    private String relativePath;
+    private int httpPort;
+    private boolean debug;
+    private ExecutorService outputExecutor;
+
+    public RawKeycloakDistribution(boolean debug, boolean manualStop) {
+        distPath = prepareDistribution();
```
Comment (pedroigor) [inline quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:67]: We should prepareDistribution only after setting the `reCreate` flag that now is removed from this PR. Could we please revert this change?

File: quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    private final List<String> errorStream = new ArrayList<>();
+    private boolean manualStop;
+    private String relativePath;
+    private int httpPort;
+    private boolean debug;
+    private boolean reCreate;
+    private ExecutorService outputExecutor;
+
+    public RawKeycloakDistribution(boolean debug, boolean manualStop, boolean reCreate) {
+        distPath = prepareDistribution();
```
Comment (pedroigor) [inline quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:68]: @andreaTP `reCreate` should be set prior to calling `prepareDistribution`. In my branch where I'm fixing this issue, I'm actually running this before starting the server.

Whatever works for you.

Comment (andreaTP) [reply quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java]: done

File: quarkus/tests/integration/pom.xml [hunk 6]
```diff
@@ -62,4 +62,39 @@
...
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <skipTests>true</skipTests>
```
Comment (pedroigor) [inline quarkus/tests/integration/pom.xml:72]: Sorry, I think I misunderstood when you mentioned that to me before. Is it really a good idea to disable tests by default?

If we are avoiding running in unit tests, perhaps we should just add a `skipQuarkusTests` similarly to the `skipTestsuite`. See https://github.com/keycloak/keycloak/blob/6a766884f632858990a05dabc4019ff3f18d972c/pom.xml#L2024.

File: quarkus/tests/pom.xml [hunk 7]
```diff
@@ -36,4 +36,18 @@
         <module>integration</module>
     </modules>
 
+    <profiles>
```
Comment (andreaTP) [inline quarkus/tests/pom.xml:39]: @pedroigor let's double-check that this works as expected.

**Review by pedroigor — APPROVED** [review]
